### ISSUE TITLE
Call to throwing default arg function allowed without throws or try in relaxed mode

### ISSUE THREAD

**Issue by cassella** [chapel-lang/chapel#13477]
### Summary of Problem

The following program compiles and runs.  I believe it should fail to
compile due to

`call to throwing function thrower without throws, try, or try!`

### Steps to Reproduce

**Source Code:**

```chapel
module M {
  config var cond = false;
  
  proc myproc(x = thrower()) {
    return x;
  }

  proc myproc2() {
    return thrower();
  }
  
  proc thrower() throws {
    if cond then
      throw new owned Error();
    return 1;
  }

  var a = myproc();

  writeln(a);
}
```

**Compile command:**
`chpl try-try-again.chpl`

**Execution command:**

```
fortytwo@magrathea:~/src/chapel (master)$ ./try-try-again -nl 1
1
fortytwo@magrathea:~/src/chapel (master)$ ./try-try-again -nl 1 --cond=true
uncaught Error: 
  try-try-again.chpl:14: thrown here
  try-try-again.chpl:18: uncaught here
```

Changing the call to `myproc()` to a call to `myproc2()` reports

```
try-try-again.chpl:8: In function 'myproc2':
try-try-again.chpl:9: error: call to throwing function thrower without throws, try, or try! (relaxed mode)
try-try-again.chpl:12: note: throwing function thrower defined here
```

So I believe the call to `myproc()` is in a context where calls to
throwing functions should require a throws, try, or try!.

**Associated Future Test(s):**

None yet.

### Configuration Information

```
chpl version 1.20.0 pre-release (304a1c8)

CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: gnu
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: native *
CHPL_LOCALE_MODEL: flat
CHPL_COMM: gasnet *
  CHPL_COMM_SUBSTRATE: udp
  CHPL_GASNET_SEGMENT: everything
CHPL_TASKS: qthreads
CHPL_LAUNCHER: amudprun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
  CHPL_NETWORK_ATOMICS: none
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_LLVM: none
CHPL_AUX_FILESYS: none

gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
```

**Comment by mppf** [comment]
@daviditen - this seems to be a related problem with error handling and default arguments. Perhaps you could look into it as a follow-up to #13437? Thanks

**Comment by daviditen** [comment]
I'll check it out.

**Comment by daviditen** [comment]
Closed with #13508.
